modules/dnn/src/model.cpp

```diff
@@ -0,0 +1,212 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+#include <algorithm>
+#include <iostream>
+#include <utility>
+#include <iterator>
+
+#include <opencv2/imgproc.hpp>
+
+namespace cv {
+namespace dnn {
+
+struct Model::Impl {
+    Size   size;
+    Scalar mean;
+    float  scale = 1.0;
+    bool   swapRB = false;
+    bool   crop = false;
+    Mat    blob;
+    std::vector<String> outNames;
+
+    void predict(Net* net, const Mat& frame, std::vector<Mat> outs) {
+        blobFromImage(frame, blob, 1.0, size, Scalar(), swapRB, crop, CV_8U);
+        net->setInput(blob, "", scale, mean);
+
+        outNames = net->getUnconnectedOutLayersNames();
+        net->forward(outs, outNames);
+    }
+};
+
+Model::Model(const std::string& model, const std::string& config)
+    : Net(readNet(model, config)), impl_(new Impl) {};
+
+Model::Model(const Net& network) : Net(network), impl_(new Impl) {};
+
+Model& Model::setInputSize(const Size& size) {
+    impl_->size = size;
+    return *this;
+}
+
+Model& Model::setInputSize(int width, int height) {
+    impl_->size = Size(width, height);
+    return *this;
+}
+
+Model& Model::setInputMean(const Scalar& mean) {
+    impl_->mean = mean;
+    return *this;
+}
+
+Model& Model::setInputScale(float scale) {
+    impl_->scale = scale;
+    return *this;
+}
+
+Model& Model::setInputCrop(bool crop) {
+    impl_->crop = crop;
+    return *this;
+}
+
+Model& Model::setInputSwapRB(bool swapRB) {
+    impl_->swapRB = swapRB;
+    return *this;
+}
+
+void Model::predict(InputArray frame, OutputArray outs) {
+    impl_->predict(this, frame.getMat(), outs.getMat());
```

Comment:
`outs` should be a vector of `cv::Mat`. So we need to use `OutputArrayOfArrays`

Reply:
done